src/views/InteractiveMap/ReactionDialog.vue

```diff
@@ -0,0 +1,401 @@
+      if (lowerBound > upperBound) {
+        return "The lower bound cannot be larger than the upper bound";
+      }
+      return true;
+    },
+    serializeMetabolites(metabolites) {
+      return metabolites
+        .filter(m => m.metabolite)
+        .map(metabolite => {
+          const compartment = metabolite.compartment
+            ? "_" + metabolite.compartment
+            : "";
+          return (
+            metabolite.stoichiometry +
+            " " +
+            this.getMetaboliteId(
+              metabolite.metabolite.id,
+              metabolite.metabolite.compartment
+            ) +
+            compartment
+          );
+        });
+    },
+    getMetaboliteId(idWithCompartment, compartmentInMetabolite) {
+      // getMetaboliteId('abc_c', 'c') => 'abc'
+      // getMetaboliteId('abc_1', 'c') => 'abc_1'
+      const compartmentInId = idWithCompartment.substring(
+        idWithCompartment.lastIndexOf("_") + 1
+      );
+      if (compartmentInId !== compartmentInMetabolite) {
+        return idWithCompartment;
+      } else {
+        return idWithCompartment.substring(
+          0,
+          idWithCompartment.lastIndexOf("_")
+        );
+      }
+    },
+    addSubstrate() {
+      this.substrates.push({
+        metabolite: null,
+        compartment: "",
+        stoichiometry: 1
+      });
+    },
+    deleteSubstrate(index) {
+      this.substrates.splice(index, 1);
+    },
+    addProduct() {
+      this.products.push({
+        metabolite: null,
+        compartment: "",
+        stoichiometry: 1
+      });
+    },
+    deleteProduct(index) {
+      this.products.splice(index, 1);
+    },
+    addReaction() {
+      if (!this.$refs.form.validate()) {
+        return;
+      }
+
+      const negativeSubstrates = this.substrates.map(m => ({
+        ...m,
+        stoichiometry: -m.stoichiometry
+      }));
+
+      const reaction: Reaction = {
+        id: this.reactionId,
+        name: this.reactionName,
+        reactionString: this.reactionString,
+        // Note: Assuming all reactions in the universal model are
+        // reversible, but this might not be the case. Could potentially use
+        // the reaction string to check reversibility.
+        lowerBound: this.lowerBound,
+        upperBound: this.upperBound,
+        metabolites: [...negativeSubstrates, ...this.products]
+          .filter(m => m.metabolite)
+          .map(m => ({
+            id: this.getMetaboliteId(m.metabolite.id, m.metabolite.compartment),
+            name: m.metabolite.name,
+            compartment: m.compartment,
+            stoichiometry: m.stoichiometry
+            // formula: null,
+            // annotation: null
+          }))
+      };
+
+      // Add the reaction to the card.
+      this.$store.commit("interactiveMap/addReaction", {
+        uuid: this.card.uuid,
+        reaction: reaction
+      });
+
+      this.$emit("simulate-card");
+      this.$refs.form.reset();
```

Comment:
You can now replace it with `this.clear()`

Reply:
Done, thanks!